metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -817,78 +812,78 @@ def _find_dataflow_recipe(
+            # For the case where we want to find a recipe for a group-by-item and metric_time, we add the time spine
+            # node (which has metric_time for all date values), and set the default join type between nodes to
+            # FULL_OUTER. This will generate a recipe where you join the node containing the group-by-item and the
+            # time spine node to get the group-by-item with metric_time.
+            candidate_nodes_for_right_side_of_join = list(self._source_node_set.source_nodes_for_group_by_item_queries)
+            candidate_nodes_for_left_side_of_join = self._select_source_nodes_with_linkable_specs(
+                linkable_specs=linkable_spec_set,
+                source_nodes=self._source_node_set.source_nodes_for_group_by_item_queries,
+            )
             default_join_type = SqlJoinType.FULL_OUTER
 
-        logger.info(f"Starting search with {len(potential_source_nodes)} potential source nodes")
+        logger.info(
+            f"Starting search with {len(candidate_nodes_for_left_side_of_join)} potential source nodes on the left "
+            f"side of the join, and {len(candidate_nodes_for_right_side_of_join)} potential nodes on the right side "
+            f"of the join."
+        )
         start_time = time.time()
 
         node_processor = PreJoinNodeProcessor(
             semantic_model_lookup=self._semantic_model_lookup,
             node_data_set_resolver=self._node_data_set_resolver,
         )
         if time_range_constraint:
-            potential_source_nodes = node_processor.add_time_range_constraint(
-                source_nodes=potential_source_nodes,
+            candidate_nodes_for_left_side_of_join = node_processor.add_time_range_constraint(
+                source_nodes=candidate_nodes_for_left_side_of_join,
                 metric_time_dimension_reference=self._metric_time_dimension_reference,
                 time_range_constraint=time_range_constraint,
             )
 
-        nodes_available_for_joins = node_processor.remove_unnecessary_nodes(
+        candidate_nodes_for_right_side_of_join = node_processor.remove_unnecessary_nodes(
             desired_linkable_specs=linkable_specs,
-            nodes=source_nodes,
+            nodes=candidate_nodes_for_right_side_of_join,
             metric_time_dimension_reference=self._metric_time_dimension_reference,
+            time_spine_node=self._source_node_set.time_spine_node,
         )
-        nodes_available_for_joins = tuple(nodes_available_for_joins)
-        if input_time_spine_source_node:
-            nodes_available_for_joins += (input_time_spine_source_node,)
         logger.info(
-            f"After removing unnecessary nodes, there are {len(nodes_available_for_joins)} nodes available for joins"
+            f"After removing unnecessary nodes, there are {len(candidate_nodes_for_right_side_of_join)} candidate "
+            f"nodes for the right side of the join"
```

Comment:
nit: f string with no variables (I'm surprised the linter didn't catch this)

Reply:
Huh. Maybe because it's multi-line? We're on an old version of Ruff, too, so it could be a false negative.

Reply:
Yeah, this is auto-formatted and f-strings without variables are done for multi-line cases for consistency.